service/history/workflow/mutable_state_impl.go

```diff
@@ -4495,6 +4495,7 @@ func (ms *MutableStateImpl) updateVersioningOverride(
 				worker_versioning.DeploymentVersionFromDeployment(d))
 		}
 	} else if ms.GetExecutionInfo().GetVersioningInfo() != nil {
+		// TODO (shahab): this behavior has changed in main branch. Update it when merging to main.
```

Comment:
not sure if this comment is still relevant

Reply:
it didn't seem it is so I removed it.

Reply:
```suggestion
```